pulumi/folio/folio.ts

```diff
@@ -0,0 +1,177 @@
+import * as request from "superagent";
+import { FolioModule } from "./classes/FolioModule";
+import * as k8s from "@pulumi/kubernetes";
+import * as eks from "@pulumi/eks";
+
+export module prepare {
+    /**
+     * Prepare a list of folio modules which can be used to deploy.
+     * @param moduleNames A list of module names to deploy.
+     * @param release The release in the form of a "quarterly" release.
+     * @returns A list of FolioModule objects.
+     */
+    export async function moduleList(moduleNames: Array<string>, release: string): Promise<FolioModule[]> {
+        var folioModules: FolioModule[] = new Array<FolioModule>();
+        console.log(`Attempting to prepare ${moduleNames.length} modules for deployment`);
+
+        const releaseModules = await getModulesForRelease(release);
+        console.log(`Got ${releaseModules.length} modules in platform-complete for ${release} release`);
+
+        for (const moduleName of moduleNames) {
+            console.log(`Trying to get module with the name: ${moduleName}`);
+            const releaseModule = getModuleByNameFromRelease(moduleName, releaseModules);
+            console.log(`Got module: ${releaseModule.id}`);
+            const start = releaseModule.id.lastIndexOf('-') + 1;
+            const end = releaseModule.id.length;
+            var moduleVersion = releaseModule.id.substring(start, end);
+            console.log(`Module version: ${moduleVersion}`);
+            var m = new FolioModule(moduleName, moduleVersion);
+            folioModules.push(m);
+        }
+
+        return folioModules;
+    }
+
+    async function getModulesForRelease(release: string): Promise<Array<any>> {
+        var endpoint = `https://raw.githubusercontent.com/folio-org/platform-complete/${release}/install.json`;
```

Comment:
Maybe you are right. I was thinking it would be better this way so we don't have to maintain a local file with the versions and just pass in a release. This would be good if we don't anticipate ever deviating from a release. But I can imagine that we'll want to apply hotfixes for a given module, which would change the version number for a given module separately from the release (maybe). So we could have version control on our local 20201-R1-release.json file to track that deviation.

Is that what you're worried about?

But I haven't looked at how hotfixes are done at all. Are they applied per-module, or for the project as a whole? If they are for the project as a whole then presumably a file like `platform-complete`'s could be generated for a hotfix that we want to apply, and we could continue to avoid hand editing the file.

Reply:
Hotfix scenario exactly what I am thinking about. I am also a little worried that the branch moves as hotfixes are developed and that it is not always guaranteed to be stable.

Reply:
Ok agreed. I'll make the change.